wled00/FX_fcn.cpp

```diff
@@ -912,7 +916,7 @@ uint32_t IRAM_ATTR_YN Segment::getPixelColor(int i) const
         return getPixelColorXY(i % vW, i / vW);
         break;
       case M12_pBar:
-        if (vStrip>0) return getPixelColorXY(vStrip - 1, vH - i -1);
+        if (vStrip>0) { i &= 0xFFFF; return getPixelColorXY(vStrip - 1, vH - i -1); }
```

Comment:
You could just go with `return getPixelColorXY(vStrip - 1, vH - (i&0xFFFF) -1);`